react/week2/todolist/src/App.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from "react";
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: [
+        {
+          id: 1,
+          description: "Get out of bed",
+        },
+        {
+          id: 2,
+          description: "Brush teeth",
+        },
+        {
+          id: 3,
+          description: "Eat breakfast",
+        },
+      ],
+    };
+  }
+
+  render() {
+    return (
+      <div>
+        <h1>todo App</h1>
+        <button onClick={this.addTodoItem}>Add Todo</button>
+        <ul>
+          {this.state.todos.map((task, i) => {
+            return (
+              <li key={i}>
+                {task.description}{" "}
+                <button data-index={i} onClick={this.deleteTodoItem}>
+                  Completed
+                </button>
+              </li>
+            );
+          })}
+        </ul>
+      </div>
+    );
+  }
+
+  addTodoItem = () => {
+    const updatedTodo = this.state.todos;
+    updatedTodo.push({
+      id: Math.floor(Math.random() * 100),
```

Comment:
Generaly, it's not a good idea to rely on small random numbers to make id like the way you did, for what if you have more than 100 todos?

Incrementation is a safe way to make `id`.
Or if you want you can use `unix timestamps` + some random number. This way you make sure there is no duplicated id in your dataset

Reply:
True,.,.I have changed  Numbers to 1000. That is math.floor(Math.random() * 1000), Thanks